### PR TITLE
feat(dsg): update code to nestjs v10

### PR DESCRIPTION
**Pull request by overbit**
Close: #6916 

## PR Details

In order to run E2E tests with the additional changes needed by `plugin-auth-core`, the e2e tests are temporary configured to use the beta version of  `plugin-auth-core` amplication/plugins#191

### Summary
🚀🧹🛠️

This pull request updates various dependencies, removes unused code, and improves the compatibility and performance of the data-service-generator package. It mainly affects the `package.json` files, the `app.module.template.ts` file, the `resolver.base.template.ts` file, the `prisma.util.ts` file, and the `prisma.service.ts` file. It also makes some minor adjustments to the Jest configuration files and the TypeScript compiler options.

> _Sing, O Muse, of the mighty pull request_
> _That updated the data-service-generator_
> _With latest versions of Jest and NestJS_
> _And removed the unused MorganModule._

### Walkthrough
*  Update Jest and related packages to the latest versions ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L243-R243), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L276-R276), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L297-R300))
*  Swap the order of `displayName` and `preset` properties in `jest.e2e.config.ts` ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-109e59a179c40674eba0dc494b7a67b9608b86e638bcee1e2c9c9c97e5ff6fe7L2-R3))
*  Remove unused dependencies on `apollo-server-express` and `nest-morgan` ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-0f8ca4854ebc0d04f3a5b0c2e2b33952568240b576a78fb49abce26afb688f31L22), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-0f8ca4854ebc0d04f3a5b0c2e2b33952568240b576a78fb49abce26afb688f31L48), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-6a0220920c91b67e25d8d92bc017b139b2a0f539a4d4b9129cc53d3ec8eddc51L3), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-6a0220920c91b67e25d8d92bc017b139b2a0f539a4d4b9129cc53d3ec8eddc51L10-R9), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-a708bad01789be26a874c5e21fc48789043ee8367e16856f5919af5229a621c9L31), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-a708bad01789be26a874c5e21fc48789043ee8367e16856f5919af5229a621c9L72), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-a708bad01789be26a874c5e21fc48789043ee8367e16856f5919af5229a621c9L140), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-cd2e1b134b0e7c78ff2c24a7ed1c6aa6d9b6286abcd488e3bb70c9c8e0c4d868L49-R50), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-a69f5a3b308d9748ebf44f2818245030a7706ed3e0a4dca0cc8945bbc3836491L2), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-d15e55f2b9c67c01bd222df8ff180f12e9abb5c17cbe79d0b0157a5b770106c1L13), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-d15e55f2b9c67c01bd222df8ff180f12e9abb5c17cbe79d0b0157a5b770106c1L116-R115))
*  Add constants and import statements for `ApolloDriver` and `ApolloDriverConfig` modules from `@nestjs/apollo` ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-a708bad01789be26a874c5e21fc48789043ee8367e16856f5919af5229a621c9R38-R40), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-a708bad01789be26a874c5e21fc48789043ee8367e16856f5919af5229a621c9L209-R211))
*  Modify the call expression for `GraphQLModule.forRootAsync` to use the new syntax and options from `@nestjs/graphql` ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-a708bad01789be26a874c5e21fc48789043ee8367e16856f5919af5229a621c9L177-R178))
*  Update the dependencies in the generated app `package.json` file ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-634b718f6fd32c0a998f0ea357ed8e53c5ce244ec01146172e2cdf9975b370deL23-R68))
*  Replace `ApolloError` with `GraphQLError` in the resolver base template ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-b79b638d637af2ecaf11adc26ec013d2b47a6e4d68812288678e95e9887ae283L2-R2), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-b79b638d637af2ecaf11adc26ec013d2b47a6e4d68812288678e95e9887ae283L105-R105), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-b79b638d637af2ecaf11adc26ec013d2b47a6e4d68812288678e95e9887ae283L122-R122))
*  Change the type annotation for the `error` parameter of the `isRecordNotFoundError` function to `any` ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-f8c6691e7443d19d02dccd1eb92e998b4c563cc609967d07add4bd88ebfeef41L4-R6))
*  Remove the `enableShutdownHooks` method from the `PrismaService` class ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-9827f2e84af7c5296a56f2a761c3e200ff39a11276fa396c9c4ecad8ff0c2f99L9-L14))
*  Update the `target` and `lib` properties to use `es2022` and `es2023` in the generated app `tsconfig.json` file ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-9a594a9d8b6600737003b91a7f5b65a97fbe3303e243b4ed8324d16e3a565924L9-R10))
*  Update the versions of the plugins used in the end-to-end tests ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-0cecb6c374ad9c2d79170d7601c6c3331e8e96c0b9d48963dfda917067083eceL9-R9), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-0cecb6c374ad9c2d79170d7601c6c3331e8e96c0b9d48963dfda917067083eceL19-R19), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-0cecb6c374ad9c2d79170d7601c6c3331e8e96c0b9d48963dfda917067083eceL29-R29), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-0cecb6c374ad9c2d79170d7601c6c3331e8e96c0b9d48963dfda917067083eceL39-R39), [link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-0cecb6c374ad9c2d79170d7601c6c3331e8e96c0b9d48963dfda917067083eceL46-R46))
*  Include the `jest.e2e.config.ts` file in the `tsconfig.spec.json` file ([link](https://github.com/amplication/amplication/pull/6921/files?diff=unified&w=0#diff-60f9191b5ce6480e59bc6f99494283e9f020fa69e595a06c0b4ccedacc0fa850R11))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/data-service-generator/package.json
Comment (mulygottlieb) [inline packages/data-service-generator/package.json:1]: Just to verify my elderly memory - there is no need to manually bump the version here (1.9.11) since it will be done by the release job, right?
I'm wondering if, for such a big change, it would be a good practice to still manually bump the version here (for contributors, etc.)
WDYT @overbit ?

File: packages/data-service-generator/src/server/static/src/prisma/prisma.service.ts [hunk 1]
```diff
@@ -6,10 +6,4 @@ export class PrismaService extends PrismaClient implements OnModuleInit {
   async onModuleInit() {
     await this.$connect();
   }
-
-  async enableShutdownHooks(app: INestApplication) {
-    this.$on("beforeExit", async () => {
-      await app.close();
-    });
-  }
```
Comment (mulygottlieb) [inline packages/data-service-generator/src/server/static/src/prisma/prisma.service.ts:14]: @overbit why was this removed? Is it just no longer needed in Prisma 5?

File: packages/data-service-generator/tests/e2e/plugins.ts [hunk 2]
```diff
@@ -36,13 +36,13 @@ export const basicAuth: PluginInstallation[] = [
     pluginId: "auth-core",
     npm: "@amplication/plugin-auth-core",
     enabled: true,
-    version: "1.1.0",
+    version: "2.0.0-beta.1",
```
Comment (mulygottlieb) [inline packages/data-service-generator/tests/e2e/plugins.ts:39]: We should add a task to remind us to change this one to "latest" as well after we publish the new auth-core.

Comment (overbit) [reply packages/data-service-generator/tests/e2e/plugins.ts:39]: I'll raise a PR straight after we merge this that will update with the v2.0.0.
I'll release the plugin v2.0.0 tomorrow with the beta tag

File: packages/data-service-generator/package.json
Comment (overbit) [reply packages/data-service-generator/package.json:1]: The current process needs the bump to happen at release time so even if it sounds like a sensible solution (to manually update it) it will affect the release process. Maybe we should avoid that version completely and evaluate the value only at release time leaving the package json version like `0.0.0-develpment`

File: packages/data-service-generator/src/server/static/src/prisma/prisma.service.ts (hunk 1 above)
Comment (overbit) [reply packages/data-service-generator/src/server/static/src/prisma/prisma.service.ts:14]: Not supported anymore and it is redundant since prisma already managed this

**Review by mulygottlieb — APPROVED** [review]
